l1-verifier/contracts/OPL1Verifier.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { IEVMVerifier } from "@ensdomains/evm-verifier/contracts/IEVMVerifier.sol";
+import { RLPReader } from "@eth-optimism/contracts-bedrock/src/libraries/rlp/RLPReader.sol";
+import { StateProof, EVMProofHelper } from "@ensdomains/evm-verifier/contracts/EVMProofHelper.sol";
+import { L1Block } from "@eth-optimism/contracts-bedrock/src/L2/L1Block.sol";
+
+struct L1WitnessData {
+    uint256 blockNo;
+    bytes blockHeader;
```

Comment:
You can also not pass blockNo as it isn't required, but kept it the same here so that we don't have to make major modification to L1Gateway api